app/apollo/resolvers/resource.updateClusterResources.js

```diff
@@ -0,0 +1,296 @@
+                var changes = null;
+                var options = {};
+                if(currentResource){
+                  // if obj already in db
+                  if (resourceHash === currentResource.hash && !hasSearchableDataChanges){
+                    // if obj in db and nothing has changed
+                    changes = {
+                      $set: { deleted: false },
+                      $currentDate: { updated: true }
+                    };
+                  }
+                  else{
+                    const toSet = { deleted: false, hash: resourceHash, data: dataStr, searchableData: searchableDataObj, searchableDataHash: searchableDataHash };
+                    if(hasSearchableDataChanges) {
+                      // if any of the searchable attrs has changes, then save a new yaml history obj (for diffing in the ui)
+                      let start = Date.now();
+                      const histId = await addResourceYamlHistObj(selfLink, dataStr);
+                      logger.info({ 'milliseconds': Date.now() - start, 'operation': 'updateClusterResources:addResourceYamlHistObj:hasSearchableDataChanges', 'data': clusterId}, 'satcon-performance');
+                      toSet['histId'] = histId;
+                    }
+                    // if obj in db and theres changes to save
+                    changes = {
+                      $set: toSet,
+                      $currentDate: { updated: true, lastModified: true },
+                      ...pushCmd
+                    };
+                  }
+                }
+                else{
+                  // adds the yaml hist item too
+                  let start = Date.now();
+                  const histId = await addResourceYamlHistObj(selfLink, dataStr);
+                  logger.info({ 'milliseconds': Date.now() - start, 'operation': 'updateClusterResources:addResourceYamlHistObj:newResource', 'data': clusterId}, 'satcon-performance');
+
+                  // if obj not in db, then adds it
+                  const total = await models.Resource.count({org_id:  org._id, deleted: false});
+                  if (total >= RESOURCE_LIMITS.MAX_TOTAL ) {
+                    throw new Error('Too many resources are registered under this organization.');
+                  }
+                  changes = {
+                    $set: { deleted: false, hash: resourceHash, histId, data: dataStr, searchableData: searchableDataObj, searchableDataHash: searchableDataHash },
+                    $currentDate: { created: true, updated: true, lastModified: true },
+                    ...pushCmd
+                  };
+                  options = { upsert: true };
+                  start = Date.now();
+                  // we need to use table.collection.updateOne(), because mongoose doesnt let you upsert when you dont set the primary key
+                  models.ResourceStat.collection.updateOne({ org_id: org._id }, { $inc: { deploymentCount: 1 } }, { upsert: true });
```

Comment:
When
- GQL code processes mutation `updateClusterResources` and
- a resource is POLLED, MODIFIED, or ADDED, and
- the resource is not already in the database,

the `deploymentCount` counter for the Org is incremented by one.

The ResourceStats related code should be removed.
- The statistic isn't _useful_
  - Nothing ever _decrements_ the `deploymentCount` counter so it doesn't represent how many resources are currently in an org.
  - The total number of resources ever added in an org, at any point in history, for any cluster, even if resource and cluster subsequently deleted, doesn't provide useful information.
- Nothing _allows_ querying the `ResourceStats` for an org (no REST or GQL endpoints).
- Nothing _requires_ querying the `ResourceStats` for an org (not show in the CLI or GUI, not logged).
- `ResourceStatSchema` includes `clusterCount` and `deployablesCount` attributes that are unused.